stellargraph/data/explorer.py

```diff
@@ -796,3 +800,202 @@ def _check_neighbourhood_sizes(self, in_size, out_size):
             self._raise_error(
                 "The number of hops for the in and out neighbourhoods must be the same."
             )
+
+
+""" Temporal random walks """
+
+
+class TemporalRandomWalk(GraphWalk):
+    """
+    Performs uniform temporal random walks on the given graph
+    """
+
+    def run(
+        self,
+        nodes=None,
+        n=None,
+        length=None,
+        bidirectional=False,
+        edge_time_label="weight",
+        seed=None,
+        step_type="uniform_random",
+    ):
+        """
+        Perform a time respecting random walk starting from the root nodes.
+
+        Args:
+            nodes: <list> The root nodes as a list of node IDs
+            n: <int> Total number of random walks per root node
+            length: <int> Maximum length of each random walk
+            bidirectional: <bool> Whether the walk extends on both direction.
+            seed: <int> Random number generator seed; default is None
+            step_type <string>: whether the next step taken in the walk is chosen uniformly at random or is biased (such as, exponential). 
+            The default is 'uniform_random', any other string value assumes a biased time respecting walk. 
+            And currently only exponentially biased random walks are implemented.  
+
+        Returns:
+            <list> List of lists of nodes ids for each of the random walks
+
+        """
+        self._check_common_parameters(nodes, n, length, seed)
+        # self._check_temporal_parameters(bidirectional, edge_time_label)
+        rs = self._get_random_state(seed)
+
+        walks = []
+        for node in nodes:  # iterate over root nodes
+            for walk_number in range(n):  # generate n walks per root node
+                walk = self._walk(
+                    node, length, bidirectional, edge_time_label, step_type, rs
+                )
+                walks.append(walk)
+        return walks
+
+    def _check_temporal_parameters(self, bidirectional, edge_time_label):
```

Comment:
This method seems to iterate through all edges in the graph to make sure that the edge weights are all positive numerical values - I think this is expensive enough that we don't want to do this up front, but instead just raise when we encounter a bad value during the random walk. I'm going to remove this.